### ISSUE TITLE
Qualified names can not be used as type constructor arguments in `case` statements.

### ISSUE THREAD

**Issue by lachenmayer** [elm-lang/elm-compiler#951]
Bar.elm:

```elm
module Bar where

type Action = Baz | Qux
```

Foo.elm:

```elm
module Foo where

import Bar exposing (Action(Qux))

type Action = NoOp | BarAction Bar.Action

foo : Action -> Int
foo action =
  case action of
    NoOp -> 0
    BarAction Bar.Baz -> 1 -- ERROR!
    BarAction Qux -> 2 -- works

faa : Bar.Action -> Int
faa action =
  case action of
    Bar.Baz -> 0 -- works
    Bar.Qux -> 1 -- works
```

Compilation result:

    Error in Foo.elm:

    Parse error at (line 11, column 18):
    unexpected "."
    expecting whitespace, a newline, pattern, cons operator '::' or arrow (->)

Expected behaviour:
`BarAction Bar.Baz` should match without error.

**Comment by sindikat** [comment]
Qualified imports are also [a problem in record updates](https://github.com/elm-lang/elm-compiler/issues/904).

**Comment by evancz** [comment]
Should be fixed in 0.17

**Comment by evancz** [comment]
Thanks for the report!
